frontend/src/components/common/SimpleTable.tsx

```diff
@@ -72,37 +73,68 @@ export default function SimpleTable(props: SimpleTableProps) {
     emptyMessage = null,
     errorMessage = null,
   } = props;
+  const location = useLocation();
+  const history = useHistory();
   const [page, setPage] = React.useState(0);
   const [currentData, setCurrentData] = React.useState(data);
   const rowsPerPageOptions = props.rowsPerPage || [5, 10, 50];
   const [rowsPerPage, setRowsPerPage] = React.useState(rowsPerPageOptions[0]);
   const classes = useTableStyle();
 
+  function addQuery(queryObj:{[key:string]: string}) {
+    const pathname = location.pathname;
+    const searchParams = new URLSearchParams(location.search);
+    for (const key in queryObj) {
+      const value = queryObj[key];
+      if (value) {
+        searchParams.set(key, value);
+      } else {
+        searchParams.delete(key);
+      }
+    }
+
+    history.push({
+      pathname: pathname,
+      search: searchParams.toString()
+    });
+  };
+
   function handleChangePage(_event: any, newPage: number) {
+    addQuery({'page': newPage.toString(), 'rowsPerPage': rowsPerPage.toString()});
     setPage(newPage);
   }
 
   function handleChangeRowsPerPage(event: any) {
+    addQuery({'page': page.toString(), 'rowsPerPage': event.target.value});
     setRowsPerPage(+event.target.value);
     setPage(0);
   }
 
   React.useEffect(() => {
-    if (currentData === data) {
-      if (page !== 0) {
-        setPage(0);
-      }
+    if (data === currentData) {
       return;
     }
-
     // If the currentData is not up to date and we are in the first page, then update
     // it directly. Otherwise it will require user's intervention.
     if ((!currentData || currentData.length === 0) || page === 0) {
       setCurrentData(data);
     }
   },
   // eslint-disable-next-line
-  [data, currentData]);
+  [data])
```

Comment:
Why did should this be removed? We need to take the value of currentdata, otherwise it will always be the first state's version.